sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -377,6 +378,29 @@ object ScalaReflection extends ScalaReflection {
           expressions.Literal.create(null, ObjectType(cls)),
           newInstance
         )
+
+      case t if isSubtype(t, localTypeOf[Enumeration#Value]) =>
+        // package example
+        // object Foo extends Enumeration {
+        //  type Foo = Value
+        //  val E1, E2 = Value
+        // }
+        // the fullName of tpe is example.Foo.Foo, but we need example.Foo so that
+        // we can call example.Foo.withName to deserialize string to enumeration.
+        val className = t.asInstanceOf[TypeRef].pre.typeSymbol.asClass.fullName
+        // this check is for spark-shell which give a default package name like '$line1.$read$$iw'
+        if (className.startsWith("$")) {
```

Comment:
I don't actually understand this limitation?

Reply:
It's actually a jline issue that default package is something like `$line32.$read.$iw.$iw.$iw.$iw.$iw`. We can't find Enum class using reflect. I choose to forbid it, but please tell me if there exists a better way to go.